### ISSUE TITLE
Custom (and callback) rules not working with custom rule messages

### ISSUE THREAD

**Issue by kskrlinnorth2** [laravel/framework#46060]
- Laravel Version: 9.45.1
- PHP Version: 8.1.0

### Description:
I am using requests for data validation. When using custom rules or callback rules along with custom messages and that rule fails, response is always full custom message array for that attribute, instead of fail message for that custom rule.

### Steps To Reproduce:
Controller method:
```
public function doSomething(MyFormRequest $request)
{
    $data = $request->validated();
    // do something with $data
}
```

Request:
```
class BookingStoreRequest extends FormRequest
{
    public function rules()
    {
        return [
            'date' => ['required', 'date_format:Y-m-d', new MyCustomDateRule],
        ];
    }

    public function messages()
    {
        return [
            'date' => [
                'required' => 'This is custom message for required rule.',
                'date_format' => 'This is custom message for date_format rule.',
                'foo' => 'bar',
            ],
        ];
    }
}
```

MyCustomDateRule:
```
class MyCustomDateRule implements Rule
{
    public function __construct()
    {
        //
    }

    public function passes($attribute, $value)
    {
        return false;
    }

    public function message()
    {
        return 'My custom rule error message.';
    }
}
```

When sending request to this controller method and all rules pass except my custom rule, I get this response:
`{"message":"This is custom message for required rule. (and 2 more errors)","errors":{"required":["This is custom message for required rule."],"date_format":["This is custom message for date_format rule."],"foo":["bar"]}}`

When sending same request, but without messages method in my request, I get actual error from my custom rule:
`{"message":"My custom rule error message.","errors":{"date":["My custom rule error message."]}}`

**Comment by driesvints** [comment]
Hey there,

Can you first please try one of the support channels below? If you can actually identify this as a bug, feel free to open up a new issue with a link to the original one and we'll gladly help you out.

- [Laracasts Forums](https://laracasts.com/discuss)
- [Laravel.io Forums](https://laravel.io/forum)
- [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
- [Discord](https://discordapp.com/invite/KxwQuKb)
- [Larachat](https://larachat.co)
- [IRC](https://web.libera.chat/?nick=laravelnewbie&channels=#laravel)

Thanks!

**Comment by kskrlinnorth2** [comment]
I described how you can simulate it and see it is bug...

**Comment by rojtjo** [comment]
You're using the incorrect syntax for your custom messages:

Change this:

```php
public function messages()
{
    return [
        'date' => [
            'required' => 'This is custom message for required rule.',
            'date_format' => 'This is custom message for date_format rule.',
            'foo' => 'bar',
        ],
    ];
}
```

To this:
```php
public function messages()
{
    return [
        'date.required' => 'This is custom message for required rule.',
        'date.date_format' => 'This is custom message for date_format rule.',
        'date.foo' => 'bar',
    ];
}
```


More info here: https://laravel.com/docs/9.x/validation#customizing-the-error-messages

**Comment by kskrlinnorth2** [comment]
> You're using the incorrect syntax for your custom messages:
> 
> Change this:
> 
> ```
> public function messages()
> {
>     return [
>         'date' => [
>             'required' => 'This is custom message for required rule.',
>             'date_format' => 'This is custom message for date_format rule.',
>             'foo' => 'bar',
>         ],
>     ];
> }
> ```
> 
> To this:
> 
> ```
> public function messages()
> {
>     return [
>         'date.required' => 'This is custom message for required rule.',
>         'date.date_format' => 'This is custom message for date_format rule.',
>         'date.foo' => 'bar',
>     ];
> }
> ```
> 
> More info here: https://laravel.com/docs/9.x/validation#customizing-the-error-messages

I think both syntaxes are correct, but I will test with this and report here.

**Comment by rojtjo** [comment]
You're right, it's actually a valid syntax but it tries to search for these options in order:

```php
["{$attribute}.{$lowerRule}", $lowerRule, $attribute]
```

Since it cannot find a combination of the attribute and the rule, or the rule itself it takes the value of the attribute, which is an array in your case. This causes all your messages to be overridden.

Source: https://github.com/laravel/framework/blob/9.x/src/Illuminate/Validation/Concerns/FormatsMessages.php#L98-L136
